src/refresh/RefreshTimer.ts

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+const MinimumBackoffInMs = 30 * 1000; // 30s
+const MaximumBackoffInMs = 10 * 60 * 1000; // 10min
+const MaxAttempts = 63;
+const JitterRatio = 0.25;
+
+export class RefreshTimer {

```

Comment:
@mrm9084 @avanigupta please take a look as I took both Python and .NET for reference when implementing the backoff part.